ember-core/shared/src/test/scala/org/http4s/ember/core/ParserSuite.scala

```diff
@@ -364,6 +364,38 @@ class ParsingSuite extends Http4sSuite {
     }
   }
 
+  test(
+    "Parser.Response.parser return everything if the body is discarded"
+  ) {
+    val defaultMaxHeaderLength = 4096
+    val raw =
+      """HTTP/1.1 200 OK
+        |Content-Length: 5
+        |
+        |helloeverything after the body""".stripMargin
+
+    val byteStream = Stream
+      .emit(raw)
+      .map(Helpers.httpifyString)
+      .through(text.utf8.encode)
+
+    for {
+      take <- Helpers.taking[IO, Byte](byteStream)
+      result <- Parser.Response
+        .parser[IO](defaultMaxHeaderLength, discardBody = true)(Array.emptyByteArray, take)
+      body <- result._1.body.through(text.utf8.decode).compile.string
+      rest <- Stream
+        .eval(result._2)
+        .flatMap(chunk => Stream.chunk(Chunk.byteVector(ByteVector(chunk.get))))
+        .through(text.utf8.decode)
+        .compile
+        .string
+    } yield {
+      assertEquals(body, "")
+      assertEquals(rest, "helloeverything after the body")
```

Comment:
I'm pretty sure we should be `assert`ing that _both_ of these are empty. Otherwise we haven't really discarded the body, just left it on the socket.

Reply:
are you sure about that? then expectNoBody would also leave data on the socket.